### PR TITLE
pull image if it does not exist on the server

### PR DESCRIPTION
**Pull request by bardiakeyvani**
in case image does not exists on the server pull image instead of failing.

### CODE REVIEW

File: dockfix.go [hunk 1]
```diff
@@ -106,6 +99,36 @@ func StartContainer(name, baseImage string) (Container, error) {
...
 
+func createContainer(baseImage string) (*docker.Container, error) {
+	dc, _ := NewClient()
+
+	exists, err := imageExists(baseImage)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		log.Print("Image does not exists. Pulling image now!")
```
Comment (idubinskiy) [inline dockfix.go]: *exist

Should probably also log `baseImage`.

File: dockfix.go [hunk 2]
```diff
@@ -106,6 +99,36 @@ func StartContainer(name, baseImage string) (Container, error) {
...
+		err = dc.PullImage(docker.PullImageOptions{Repository: baseImage}, docker.AuthConfiguration{})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	log.Print("Creating new container for ", baseImage)
+	cont, err := dc.CreateContainer(
```
Comment (idubinskiy) [inline dockfix.go]: You can just return here instead of assigning.

File: dockfix.go [hunk 3]
```diff
@@ -127,3 +150,23 @@ func RemoveContainer(c Container) error {
...
+
+func imageExists(baseImage string) (bool, error) {
+	dc, _ := NewClient()
+	imageName := strings.Split(baseImage, ":")[0]
+	images, err := dc.ListImages(docker.ListImagesOptions{Filter: imageName})
+	if err != nil {
+		return false, err
+	}
+
+	if len(images) > 0 {
```
Comment (idubinskiy) [inline dockfix.go]: This isn't really needed. If length is 0, the range loop just won't do anything.

**Comment by bardiakeyvani** [comment]
update 70ba2a79e141e75b346930a4f0190e1fd650f396

**Comment by idubinskiy** [comment]
:+1:

**Pull request merged**